tests/utils/test_urlutils.py

```diff
@@ -524,3 +524,63 @@ def test_same_domain_invalid_url(url1, url2):
     """Test same_domain with invalid URLs."""
     with pytest.raises(ValueError):
         urlutils.same_domain(QUrl(url1), QUrl(url2))
+
+@pytest.mark.parametrize('url, incdec, output', [
```

Comment:
I'm fine with perfectionism, feel free to point out the stuff I need to change :smile: